### PR TITLE
DEV: Convert min_trust_level_to_create_tag to groups

### PR DESCRIPTION
**Pull request by Drenmi**
[Meta](https://meta.discourse.org/t/283408)

### What is this change?

We're changing the implementation of trust levels to use groups. Part of this is to have site settings that reference trust levels use groups instead. It converts the `min_trust_level_to_create_tag ` site setting to `create_tag_allowed_groups`.

This PR maintains backwards compatibility until we can update plugins and themes using this.

### CODE REVIEW

File: config/site_settings.yml
Comment (Drenmi) [inline config/site_settings.yml:2974]: Admins, staff, TL3.

File: lib/guardian/tag_guardian.rb
Comment (tgxworld) [inline lib/guardian/tag_guardian.rb:12]: Since this is core's code, why do we still reference the deprecated site setting here?

File: lib/site_settings/deprecated_settings.rb
Comment (tgxworld) [inline lib/site_settings/deprecated_settings.rb:36]: I wonder if there is a way for us to map the setting of `min_trust_to_create_tag` to `create_tag_allowed_groups` so that we don't have to keep `min_trust_to_create_tag` around and have to check two site settings whenever we want to check if a user can create a tag.

File: lib/guardian/tag_guardian.rb
Comment (oblakeerickson) [reply lib/guardian/tag_guardian.rb:12]: From the PR description I think he is trying to maintain backwards compatibility with plugins until they are all updated.

File: lib/site_settings/deprecated_settings.rb
Comment (martin-brennan) [reply lib/site_settings/deprecated_settings.rb:36]: No there's not; we are happy with the process we are following here. Certain settings will have to be both checked until all the plugins are all migrated across, we have done this many times now and it's only temporary.

File: lib/guardian/tag_guardian.rb
Comment (martin-brennan) [reply lib/guardian/tag_guardian.rb:12]: Yes this is only necessary very occasionally for settings that are more widely used and is only temporary. Many of these settings moves only needed core changes.

File: config/site_settings.yml
Comment (martin-brennan) [reply config/site_settings.yml:2981]: I don't _think_ this is necessary because admin and staff will already be in the TL3 auto group; however I also don't think it's a bad thing to do either since the old setting was `TrustLevelAndStaffSetting` :+1:

**Review by martin-brennan — APPROVED** [review]
LGTM, thanks Ted :+1:

File: lib/site_settings/deprecated_settings.rb
Comment (tgxworld) [reply lib/site_settings/deprecated_settings.rb:36]: That doesn't seem to be safe. If you allow writes to both site settings, how are we going to reconcile it when it is time to drop the old site setting?

**Comment by martin-brennan** [comment]
We discussed internally and we want to do a couple of things:

1. Change the migration to a post migration to avoid potential data issues when deploying this change
2. Look into how we can automatically map the old deprecated setting to the new one via `SiteSettings::DeprecatedSettings` (I am doing this now)

Until then we should hold off on merging this particular PR.

File: config/site_settings.yml
Comment (Drenmi) [reply config/site_settings.yml:2981]: I suspect you are correct. The reason I did this is so we can lean maximally on existing tests, which will fail without this. Making too many changes to both tests and code at once always feels a bit shady. 😅

**Pull request merged**
